streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -992,7 +992,7 @@ private StreamThread createStreamThread(final long cacheSizePerThread, final int
         if (isRunningOrRebalancing()) {
             synchronized (changeThreadCount) {
                 for (final StreamThread streamThread : threads) {
-                    if (streamThread.isAlive()) {
+                    if (streamThread.isAlive() && (!streamThread.getName().equals(Thread.currentThread().getName()) || threads.size() == 1)) {
```

Comment:
I guess technically this might not work if we have only one live thread left, but other dead threads still in the list -- in that case we might skip over the live thread when we really should have removed it. I'm not sure how much of a problem this would really be, just wanted to point it out.

Reply:
We don't keep the dead threads in this list. When a thread dies we remove it form the list and updated the number of dead threads metric.

Reply:
Then why check `isAlive()` here at all? (I'm not necessarily requesting changes here, just trying to understand)

Reply:
there is a very small window between when thread dies with the old handler and when it is removed from the the list. So we have to check to make sure for now